### PR TITLE
fix constructor argument types

### PR DESCRIPTION
**Pull request by xabbuh**

### CODE REVIEW

File: src/Node/Expression/BlockReferenceExpression.php [hunk 1]
```diff
@@ -22,7 +22,10 @@
  */
 class BlockReferenceExpression extends AbstractExpression
 {
-    public function __construct(\Twig_NodeInterface $name, ?Node $template, $lineno, $tag = null)
+    /**
+     * @param Node|null $template
+     */
+    public function __construct(\Twig_NodeInterface $name, $template, $lineno, $tag = null)
```
Comment (xabbuh) [inline src/Node/Expression/BlockReferenceExpression.php:28]: The type hint can be added in `2.x` (see the legacy `is_bool()` check below).

File: src/Node/ForNode.php [hunk 2]
```diff
@@ -25,7 +25,7 @@ class ForNode extends Node
 {
     protected $loop;
 
-    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, ?AbstractExpression $ifexpr, ?\Twig_NodeInterface $body, ?\Twig_NodeInterface $else, $lineno, $tag = null)
+    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, ?AbstractExpression $ifexpr, \Twig_NodeInterface $body, ?\Twig_NodeInterface $else, $lineno, $tag = null)
```
Comment (xabbuh) [inline src/Node/ForNode.php:28]: the body was not nullable before #3331

File: src/Node/IfNode.php [hunk 3]
```diff
@@ -21,7 +21,7 @@
  */
 class IfNode extends Node
 {
-    public function __construct(?\Twig_NodeInterface $tests, ?\Twig_NodeInterface $else, $lineno, $tag = null)
+    public function __construct(\Twig_NodeInterface $tests, ?\Twig_NodeInterface $else, $lineno, $tag = null)
```
Comment (xabbuh) [inline src/Node/IfNode.php:24]: `$tests` were not nullable before #3331

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Node/Expression/BlockReferenceExpression.php (hunk 1 above)
Comment (stof) [reply src/Node/Expression/BlockReferenceExpression.php:28]: could even be added in 1.x as the min PHP version has been bumped.

**Pull request merged**